sdks/python/apache_beam/runners/worker/worker_pool_main.py

```diff
@@ -47,21 +47,26 @@
 class BeamFnExternalWorkerPoolServicer(
     beam_fn_api_pb2_grpc.BeamFnExternalWorkerPoolServicer):
 
-  def __init__(self, worker_threads, use_process=False,
-               container_executable=None):
+  def __init__(self, worker_threads,
+               use_process=False,
+               container_executable=None,
+               state_cache_size=0):
     self._worker_threads = worker_threads
     self._use_process = use_process
     self._container_executable = container_executable
+    self._state_cache_size = state_cache_size
     self._worker_processes = {}
 
   @classmethod
   def start(cls, worker_threads=1, use_process=False, port=0,
-            container_executable=None):
+            state_cache_size=0, container_executable=None):
     worker_server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
     worker_address = 'localhost:%s' % worker_server.add_insecure_port(
         '[::]:%s' % port)
-    worker_pool = cls(worker_threads, use_process=use_process,
-                      container_executable=container_executable)
+    worker_pool = cls(worker_threads,
+                      use_process=use_process,
+                      container_executable=container_executable,
+                      state_cache_size=state_cache_size)
```

Comment:
This should not occur here but come from the pipeline options from the provisioning endpoint.

Reply:
We need this here as well because this class is also used for testing. It will be overridden for the non-testing case.